src/masternode.cpp

```diff
@@ -466,6 +456,63 @@ bool CMasternodeBroadcast::Create(CTxIn txin, CService service, CKey keyCollater
     return true;
 }
 
+bool CMasternodeBroadcast::CheckDefaultPort(std::string strService, std::string& strErrorRet, std::string strContext)
+{
+    CService service = CService(strService);
+    int defaultPort = 0;
+
+    // Get default port for current network and check against currently used port
+    switch (Params().NetworkID()) {
+        // Main network
+        case CBaseChainParams::MAIN:
+            defaultPort = Params(CBaseChainParams::MAIN).GetDefaultPort();
```

Comment:
Is there a reason to use a switch and not something like `defaultPort = Params().GetDefaultPort();` ?

Reply:
Mainly because of the different error texts (in fact I found the error because of the (wrong-) identical texts), but I think it's possible to solve this more elegant.